concordium-std-derive/src/lib.rs

```diff
@@ -1202,6 +1224,154 @@ fn schema_type_fields(fields: &syn::Fields) -> syn::Result<proc_macro2::TokenStr
     }
 }
 
+/// We reserve a number of error codes for custom errors, such as ParseError,
+/// that are provided by concordium-std. The error codes for user-defined error
+/// types will have this number as an offset.
+const RESERVED_ERROR_CODES: u32 = 100;
+
+/// Derive the conversion of enums that represent error types into the Reject
+/// struct which can be used as the error type of init and receive functions.
+/// Creating custom enums for error types can provide meaningful error messages
+/// to the user of the smart contract.
+///
+/// Please note:
+///   - At the moment, we can only derive fieldless enums.
+///   - The error-type enum needs to derive (or implement) the Copy and Clone
+///     traits.
```

Comment:
I'm guessing it should be possible to not require Copy. Practically it is not a problem, since fieldless enums are easily copyable, but if we can avoid requiring it we can avoid annoying errors for the users.